### DISCUSSION TITLE
feat: dont show errors if plugin isn't supported on the current platform

### DISCUSSION THREAD

**Discussion by josh-m-sharpe** [ionic-team/capacitor#2959] [Ideas]
It seems like capacitor should be able to handle exceptions like the following and prevent us from having to wrap calls like this in platform checks.

<img width="625" alt="Screen Shot 2020-05-21 at 7 47 34 AM" src="https://user-images.githubusercontent.com/39473/82556358-9267bc80-9b37-11ea-95cf-056ba5d7b8de.png">

More generally, if a native capacitor plugin isn't supported on [some platform]:
1) show a warning
2) resolve true

**Comment by jcesarmobile** [comment]
Added the needs discussion tag as should be discussed

BTW, there is a `isPluginAvailable` method you can use instead of platforms checks, so you don't need to know if it's supported by the platform beforehand. Not ideal neither, but more future proof in case it gets implemented in the future

**Comment by indraraj26** [comment]
instead of error just throw console.warn that this plugin doesnt support for web

**Reply by imhoffd** [reply]
I believe this is what Cordova does, correct?

**Comment by josh-m-sharpe** [comment]
I'd suggest implementing some kind of no-op implementation in the platform where the plugin isn't supported.  The warning could be added there.  Then you can probably pitch any platform-detection logic.

**Comment by aparajita** [comment]
I have to agree with the others. Most of the time there is no harm in making the call a no-op on the web, so that should be the default.
